src/integrationTests/uid2SecureSignaltest.ts

```diff
@@ -0,0 +1,192 @@
+    return;
+  });
+});
+
+afterEach(() => {
+  consoleWarnMock.mockRestore();
+  getAdvertisingTokenMock.mockRestore;
+  secureSignalProvidersPushMock.mockRestore();
+  window.getUid2AdvertisingToken = undefined;
+});
+
+describe("when use script without SDK integrated", () => {
+  describe("when getUid2AdvertisingToken exists and returns valid advertisingToken", () => {
+    test("should send signal to Google ESP", async () => {
+      window.getUid2AdvertisingToken = getAdvertisingTokenMock;
+      getAdvertisingTokenMock.mockReturnValue("testToken");
+      uid2ESP = new Uid2SecureSignalProvider();
+      //@ts-ignore
+      expect(uid2ESP.retrieveAdvertisingTokenHandler()!()).toBe("testToken");
+      expect(secureSignalProvidersPushMock).toHaveBeenCalledTimes(1);
+      await expect(secureSignalProvidersPushMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: "uidapi.com",
+        })
+      );
+      expect(secureSignalProvidersResolveMock).toHaveBeenCalledWith("testToken")
+    });
+  });
+
+  describe("when getUid2AdvertisingToken is not definied", () => {
+    test("should not send signal to ESP", () => {
+      uid2ESP = new Uid2SecureSignalProvider();
+      expect(secureSignalProvidersPushMock).not.toBeCalled();
+    });
+
+    describe("when publisher trigger registerSecureSignalProvider", () => {
+      test("should log warning message to console and not send message", () => {
+        uid2ESP.registerSecureSignalProvider();
+        expect(console.warn).toHaveBeenCalledTimes(1);
+        expect(consoleWarnMock).toHaveBeenCalledWith(
+          "Please implement `getUid2AdvertisingToken`"
+        );
+        expect(secureSignalProvidersPushMock).not.toBeCalled();
+      });
+    });
+  });
+
+  describe("when getUid2AdvertisingToken exists and returns invalid token", () => {
+    test("should not send signal to ESP", () => {
+      getAdvertisingTokenMock.mockReturnValue("");
+      new Uid2SecureSignalProvider();
+      expect(secureSignalProvidersPushMock).not.toBeCalled();
+    });
+  });
+});
+
+describe("when use script with SDK", () => {
+  const refreshFrom = Date.now() + 100;
+  const identity = mocks.makeIdentityV2({ refresh_from: refreshFrom });
+  const refreshedIdentity = mocks.makeIdentityV2({ advertising_token: 'refreshed_token' })
+  let uid2: UID2;
+  beforeEach(() => {
+    uid2ESP = new Uid2SecureSignalProvider();
+    uid2 = new UID2();
+    window.__uid2 = uid2
+  })
+
+  afterEach(() => {
+    //@ts-ignore
+    window.__uid2Esp = undefined
+  })
+        
+  describe("when SDK enable esp with identity", () => {
+    describe("When script loaded before SDK loaded", () => {
+      test("should send signal to Google ESP when SDK initialized", async () => {
+        uid2.init({ identity, enableSecureSignals: true });
+        //@ts-ignore
+        expect(uid2ESP.retrieveAdvertisingTokenHandler()!()).toBe(identity.advertising_token);
```

Comment:
It would be better to make this check the identity the push mock is called with. This `expect` tests a private method, so the test might break if you do an internal refactor, and it also doesn't test that the token is actually passed to Secure Signals.

Reply:
Yes, this line here is to test the `retrieveAdvertisingTokenHandler` is calling the correct function, in this case, it should use `__uid2.getAdverstingToken` as `retrieveAdvertisingTokenHandler`
At line 141 `expect(secureSignalProvidersResolveMock).toHaveBeenCalledWith(identity.advertising_token)` is checking if the token is passed to SecureSignal as the push mock is defined as this  `secureSignalProvidersPushMock = jest.fn(async (p: EncryptedSignalProvider) => {
    secureSignalProvidersResolveMock(await p.collectorFunction())
  })`